app/knight_class.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight_data: dict) -> None:
+        self.name = knight_data.get("name")
+        self.power = self.calculation_character(knight_data, "power")
+        self.hp = self.calculation_character(knight_data, "hp")
+        self.protection = self.calculation_character(knight_data, "protection")
+
+    def calculation_character(

```

Comment:
Use verbs in the method names, name should represent action in this case
```suggestion
    def calculate_character(

```